Resources/public/js/views/services/plugins/ez-versionsloadplugin.js

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-versionsloadplugin', function (Y) {
+    "use strict";
+    /**
+     * Provides the versions list load plugin
+     *
+     * @module ez-versionsloadplugin
+     */
+    Y.namespace('eZ.Plugin');
+
+    /**
+     * Object versions load plugin. It sets an event handler to load versions
+     * of content for versions tab in location view.
+     *
+     * In order to use it you need to fire `loadVersions` event with parameter
+     * `content` containing the eZ.Content object for which you want to load versions.
+     *
+     * @namespace eZ.Plugin
+     * @class VersionsLoad
+     * @constructor
+     * @extends eZ.Plugin.ViewServiceBase
+     */
+    Y.eZ.Plugin.VersionsLoad = Y.Base.create('versionsloadplugin', Y.eZ.Plugin.ViewServiceBase, [], {
+
+        initializer: function () {
+            this.onHostEvent('*:loadVersions', this._loadVersions);
+        },
+
+        /**
+         * Loads versions list for content given in event facade. When loading of versions
+         * is finished, then it is set in `versions` attribute of the event target.
+         *
+         * @method _loadVersions
+         * @private
+         * @param {EventFacade} e loadVersions event facade
+         * @param {eZ.Content} e.content the content
+         *
+         */
+        _loadVersions: function (e) {
+            var service = this.get('host'),
+                capi = service.get('capi'),
+                options = {
+                    api: capi,
+                };
+
+            e.content.loadVersions(options, function (error, versions) {
+                if (error) {
+                    e.target.set('loadingError', true);
+                } else {
+                    e.target.set('versions', versions);
```

Comment:
I think you also need to set `loadingError` to `false` for the case where an error happened previously and the user hits the retry button.